lua/nvim-treesitter/refactor/smart_rename.lua

```diff
@@ -43,6 +43,7 @@ function M.smart_rename(bufnr)
     table.insert(edits, text_edit)
   end
   vim.lsp.util.apply_text_edits(edits, bufnr)
+  ts_utils.goto_node(node_at_point, false, true)
```

Comment:
Can you do that before the edits are applied just to be sure the node is valid ?

Reply:
It's lua so the node shouldn't be garbage collected though it's already invalid. But it is still not completely correct because due to renaming of the same reference in the same line the node position might have changed.